src/Campaigns/Routes/Campaign.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Give\Campaigns\Routes;
+
+use Exception;
+use Give\API\RestRoute;
+use Give\Campaigns\Models\Campaign as CampaignModel;
+use Give\Campaigns\ValueObjects\CampaignStatus;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class Campaign implements RestRoute
+{
+
+    /** @var string */
+    protected $endpoint = 'campaigns/(?P<id>[0-9]+)';
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute()
+    {
+        // Get campaign
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => [$this, 'handleGetRequest'],
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                        'validate_callback' => function ($id) {
+                            return filter_var($id, FILTER_VALIDATE_INT);
```

Comment:
You don't need a callback here. The rest api takes care of this by default. 

Check out: https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/